src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -187,13 +181,37 @@ We support
                         });
                     }
                 }
+
+                IConfigurationSection allocationSection = configurationSection.GetSection(nameof(FeatureDefinition.Allocation));
+
+                if (allocationSection.Exists())
+                {
+                    allocation = new Allocation();
+                    allocationSection.Bind(allocation);
```

Comment:
I'd like for us to avoid bind and instantiate our objects ourselves without reflection needed.